### PR TITLE
Cluster operator status

### PR DESCRIPTION
**Pull request by enxebre**
Move from old operatorstatus.openshift.io to Cluster Operator object

### CODE REVIEW

File: pkg/operator/status.go [hunk 1]
```diff
@@ -1,61 +1,49 @@
...
-	if cond.Type == v1.OperatorStatusConditionTypeDegraded {
-		return fmt.Errorf("invalid cond %s", cond.Type)
-	}
-
-	// TODO(yifan): Fill in the Extention field for the status
+//syncStatus applies the new condition to the mao ClusterOperator object.
+func (optr *Operator) syncStatus(cond osconfigv1.ClusterOperatorStatusCondition) error {
 	// to report the status of all the managed components.
-	status := &v1.OperatorStatus{
+	clusterOperator := &osconfigv1.ClusterOperator{
```
Comment (ingvagabund) [inline pkg/operator/status.go]: What's the relation between mao operator status and cluster operator status? Is cluster operator an abstract way how to update status? Or WIP at the moment? If anything, I would expect `osconfigv1.Operator` struct instead for generic status updates.

Hmm, mao cluster operator. What do I need to read to understand why is the `cluster` part of mao operator name? Assuming it means a cluster operator is a name for a class of operators, right? On the cluster level?

File: pkg/operator/sync.go [hunk 2]
```diff
@@ -39,8 +40,18 @@ func (optr *Operator) syncAll(config OperatorConfig) error {
 	}
 	glog.Info("Synched up cluster api controller")
 
-	return optr.syncStatus(v1.OperatorStatusCondition{
-		Type:    v1.OperatorStatusConditionTypeDone,
+	// TODO(alberto): create funcs to for atomic conds changes
```
Comment (ingvagabund) [inline pkg/operator/sync.go]: You can make the `syncStatus` to accept variable number of arguments, resp. a list with variable size.

File: pkg/operator/status.go [hunk 3]
```diff
@@ -1,61 +1,49 @@
...
 
-// syncStatus applies the new condition to the mco's OperatorStatus object.
-func (optr *Operator) syncStatus(cond v1.OperatorStatusCondition) error {
-	if cond.Type == v1.OperatorStatusConditionTypeDegraded {
-		return fmt.Errorf("invalid cond %s", cond.Type)
-	}
-
-	// TODO(yifan): Fill in the Extention field for the status
+//syncStatus applies the new condition to the mao ClusterOperator object.
+func (optr *Operator) syncStatus(cond osconfigv1.ClusterOperatorStatusCondition) error {
```
Comment (ingvagabund) [inline pkg/operator/status.go]: Not sure if this is going to work but worth trying :) 
```go
func (optr *Operator) syncStatus(statusConditions ...osconfigv1.ClusterOperatorStatusCondition) error {
	// to report the status of all the managed components.
	clusterOperator := &osconfigv1.ClusterOperator{
		ObjectMeta: metav1.ObjectMeta{
			Namespace: optr.namespace,
			Name:      optr.name,
		},
		Status: osconfigv1.ClusterOperatorStatus{
			Version: version.Raw,
		},
	}
	for _, cond := range statusConditions {
		cvoresourcemerge.SetOperatorStatusCondition(&clusterOperator.Status.Conditions, cond)
	}
	_, _, err := ApplyClusterOperator(optr.osClient.ConfigV1(), clusterOperator)
	return err
}
```

File: pkg/operator/sync.go (hunk 2 above)
Comment (enxebre) [reply pkg/operator/sync.go]: Yeh, this is a first step to have the right CRD object in place intentionally simple, hence the TODO. Once we get clarity on the CVO expectations for updgrades we can model some functions

**Comment by ingvagabund** [comment]
Ok, fair enough
/approve

**Pull request merged**